src/v/cloud_storage/tests/s3_imposter.cc

```diff
@@ -512,7 +512,7 @@ enable_cloud_storage_fixture::enable_cloud_storage_fixture() {
         cfg.cloud_storage_region.set_value(
           std::optional<ss::sstring>{"us-east1"});
         cfg.cloud_storage_bucket.set_value(
-          std::optional<ss::sstring>{test_bucket_name()});
+          std::optional<ss::sstring>{random_test_bucket_name()()});
```

Comment:
is there a reason why this is needed? ease of debugging?

Reply:
Ease of debugging was my primary motivation. I don't think it makes a practical difference either way, but if there is a good reason to not randomize bucket names between `s3_imposter` instances we can revert this.